src/core/server/saved_objects/service/lib/saved_object_aggs_types/bucket_aggs/index.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as rt from 'io-ts';
+
+import { FieldBasicRT } from '../helpers';
+
+export const BucketAggsTypeRt = rt.partial({
+  filter: rt.type({
+    term: rt.record(rt.string, rt.string),
+  }),
+  histogram: rt.intersection([
+    FieldBasicRT,
+    rt.type({ interval: rt.number }),
+    rt.partial({
+      min_doc_count: rt.number,
+      extended_bounds: rt.type({ min: rt.number, max: rt.number }),
+      keyed: rt.boolean,
+      missing: rt.number,
+      order: rt.record(rt.string, rt.literal('asc', 'desc')),
+    }),
+  ]),
+  terms: rt.intersection([
+    FieldBasicRT,
+    rt.partial({
+      field: rt.string,
+      size: rt.number,
+      show_term_doc_count_error: rt.boolean,
+      order: rt.record(rt.string, rt.literal('asc', 'desc')),
+    }),
```

Comment:
> I am using io-ts to validate the schema since io-ts allow recursive type and aggregations are recursive

Do we have a practical use case where this recursive schema feature would be useful to us?

Reply:
I think we will, for example, we can use a filter aggregations and then do a cardinality on this filter and the same thing with top_hits aggregations. I think that's the power of aggregation in elastic search and we do not want to limited.
I can see us using the top_hits aggregations for the case solutions for our tags.

Reply:
Fair enough. That's a shame we are forced to use something else than `config-schema` for that, however TBH that would need a really heavy refactoring to allow recursive typings in our library, if even doable.